src/javascript/app/pages/trade/duration.js

```diff
@@ -180,11 +180,26 @@ const Durations = (() => {
         return (value * seconds[from_unit]) / seconds[to_unit];
     };
 
+    const getSmallestDuration = () => {
+        const unit               = CommonFunctions.getElementById('duration_units');
```

Comment:
I'm not sure if it's a good idea to get this from DOM when you can get it from something like `list` as above on line 134. we can turn that into a global variable or store the smallest value and return it here.

Reply:
I think we don't really need this function because we don't need `unit_name` and `unit_value` too.

Reply:
I'm not sure how we can compare the values without checking the value from DOM. This change compare the current value of DOM and local storage, and returns the latest (higher value) of Date.

Previously if you go to Volatility(today's date) and go to Forex tab, the date barrier sets to current date and it caused an error.

Reply:
you can look at `list` above as mentioned, it already has this list of values stored.

Reply:
oh ok got it 👍